lib/yt/models/snippet.rb

```diff
@@ -51,6 +51,17 @@ def can_reply?
         @can_reply ||= data.fetch 'canReply', false
       end
 
+      def top_level_comment
+        if @top_level_comment.nil?
+          d = data.fetch('topLevelComment', nil).tap do |data|
+            data[:id] = data.fetch 'id', nil
+            data[:snippet] = data.fetch 'snippet', nil
+          end
+          @top_level_comment = Yt::Comment.new(d) if d
+        end
+        @top_level_comment
+      end
```

Comment:
As of side cases, I have to used ActiveSupport::HashWithIndifferentAccess to wrap `data['topLevelComment']` since the initializer of `Yt::Comment` requires all keys as symbols, e.g. `data[:snippet]`